src/wasm32/ffi.c

```diff
@@ -204,64 +219,121 @@ ffi_call, (ffi_cif * cif, ffi_fp fn, void *rvalue, void **avalue),
 
   var orig_stack_ptr = stackSave();
   var structs_addr = orig_stack_ptr;
+  var varargs;
+  if (nfixedargs != nargs) {
+    var size = 0;
+    for (var i = nfixedargs; i < nargs; i++) {
+      var arg_unboxed = unbox_small_structs(DEREF_U32(arg_types, i));
+      var arg_type = arg_unboxed[0];
+      var item = ffi_struct_size_and_alignment(arg_type);
+      var item_size = item[0];
+      size += item_size;
+    }
+    varargs = stackAlloc(size);
+  }
+
```

Comment:
I think this would be better as:
```suggestion
  if (nfixedargs !== nargs) {
    for (var i = nfixedargs; i < nargs; i++) {
      var arg_unboxed = unbox_small_structs(DEREF_U32(arg_types, i));
      var arg_type = arg_unboxed[0];
      var item = ffi_struct_size_and_alignment(arg_type);
      var item_size = item[0];
      structs_addr -= item_size;
      var arg_ptr = DEREF_U32(avalue, i);
	  HEAP8.subarray(structs_addr, structs_addr + item_size)
	  	   .set(HEAP8.subarray(arg_ptr, arg_ptr + item_size));
    }
  }
```